### PR TITLE
feat/US34084-Create DAO UI Wizard Styles

### PR DESCRIPTION
**Pull request by roshan-bisht-lab49**
**Description**:
Screenshot 2023-05-24 at 1 17 15 PM
Screenshot 2023-05-24 at 1 17 26 PM
Screenshot 2023-05-24 at 1 17 36 PM
Screenshot 2023-05-24 at 1 18 04 PM
Screenshot 2023-05-24 at 1 19 08 PM

**Also Updated the new Design for Token DAO / NFT DAO**:

Screenshot 2023-05-26 at 8 51 55 PM
Screenshot 2023-05-26 at 8 49 35 PM
Screenshot 2023-05-26 at 8 45 28 PM
Screenshot 2023-05-25 at 4 10 33 PM

### CODE REVIEW

**Comment by roshan-bisht-lab49** [comment]
@damiano-melcarne-lab49  I have a detailed discussion with Tunde regarding the read only inputs she is working on it so until she comes up with a design for the ready only inputs I was thinking above screenshots UI might be the better way to show the disabled fields. let me know your thoughts on the new design.

File: src/dex-ui-components/base/FormInputList/FormInputList.tsx [hunk 1]
```diff
@@ -23,7 +26,17 @@ interface FormInputListProps<FormType extends FieldValues, FormPath extends Arra
...
 ) {
-  const { fields, defaultFieldValue, formPath, register, append, remove } = props;
+  const {
+    fields,
+    defaultFieldValue,
+    formPath,
+    register,
+    append,
+    remove,
+    fieldLabel = "Member wallet address",
```
Comment (damiano-melcarne-lab49) [inline src/dex-ui-components/base/FormInputList/FormInputList.tsx:36]: This should be a generic component so we don't want to use defaults for these fields. These values need to be moved to the  `FormInputList` in the `MultiSigDAOGovernanceForm`.

File: src/dex-ui-components/base/FormInputList/FormInputList.tsx [hunk 2]
```diff
@@ -36,26 +49,32 @@ export function FormInputList<FormType extends FieldValues, FormPath extends Arr
...
-              Remove
-            </Button>
+            <IconButton
+              size="xs"
+              variant="link"
+              marginLeft="0.25rem"
+              marginBottom="1.2rem"
+              aria-label="remove-multisig-member"
+              onClick={() => handleRemoveInputClicked(index)}
+              icon={<CancelledStepIcon fill={Color.Grey_Blue._500} options={{ height: 21, width: 21 }} />}
```
Comment (damiano-melcarne-lab49) [inline src/dex-ui-components/base/FormInputList/FormInputList.tsx:70]: I recommend updating the `CancelledStepIcon` to use the `createIcon` function from `@chakra-ui/react`. This allows the icon to inherit all of the ChakraUI styling options. You will be able to directly style the components without using intermediary props, e.g. `<CancelledStepIcon boxSize="4">`.

File: src/dex-ui-components/base/Icons/DefaultLogoIcon.tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import { Icon } from "@chakra-ui/react";
```
Comment (damiano-melcarne-lab49) [inline src/dex-ui-components/base/Icons/DefaultLogoIcon.tsx:1]: Use the `createIcon` approach for new Icons (the reasoning for this is in an early comment).

File: src/dex-ui/pages/dao/CreateADAO/forms/DAODetailsForm.tsx [hunk 4]
```diff
@@ -27,36 +57,43 @@ export function DAODetailsForm() {
...
-        <FormInput<"description">
+      <FormTextArea<"description">
+        textAreaProps={{
+          id: "description",
+          label: "Description",
+          placeholder: "Add a description for your DAO",
+          register: {
+            ...register("description", {
+              required: { value: true, message: "A description is required." },
+              validate: (value) => value.length > 239 || "At-least 240 characters is required.",
```
Comment (damiano-melcarne-lab49) [inline src/dex-ui/pages/dao/CreateADAO/forms/DAODetailsForm.tsx:68]: 1) `At-least` => `At least`
2) The 240 character count is a `max` character limit not a `min`.

File: src/dex-ui/pages/dao/CreateADAO/forms/MultiSigDAOGovernanceForm.tsx [hunk 5]
```diff
@@ -22,9 +36,12 @@ export function MultiSigDAOGovernanceForm() {
           label: "Admin wallet address",
           type: "text",
           placeholder: "Enter address",
+          unit: getAdminAddressStatusIcon(),
           register: {
             ...register("governance.admin", {
-              required: { value: true, message: "An admin wallet address is required." },
+              required: { value: true, message: "Admin wallet address is required." },
+              validate: (value) => checkForValidAccountId(value) || "Invalid address, please, enter a different one.",
+              onChange: debounce(handleAdminAddressChange, 500),
```
Comment (damiano-melcarne-lab49) [inline src/dex-ui/pages/dao/CreateADAO/forms/MultiSigDAOGovernanceForm.tsx:44]: Store the debounce millisecond time in a constant. We should have a consistent delay for these validations across the application.

File: src/dex-ui/utils/utils.ts [hunk 6]
```diff
@@ -19,3 +21,11 @@ export function convertToByte32(hexString: string): Uint8Array {
 export function checkForValidTokenId(tokenId: string): boolean {
   return /^\d+\.\d+\.\d+$/.test(tokenId);
 }
+
+export function checkForValidAccountId(accountId: string): boolean {
+  try {
+    return !isNil(AccountId.fromString(accountId));
```
Comment (damiano-melcarne-lab49) [inline src/dex-ui/utils/utils.ts:27]: ramda also has a `isNotNil` function

**Review by damiano-melcarne-lab49 — CHANGES_REQUESTED** [review]

File: src/dex-ui-components/base/FormInputList/FormInputList.tsx (hunk 1 above)
Comment (roshan-bisht-lab49) [reply src/dex-ui-components/base/FormInputList/FormInputList.tsx]: Done

File: src/dex-ui-components/base/FormInputList/FormInputList.tsx (hunk 2 above)
Comment (roshan-bisht-lab49) [reply src/dex-ui-components/base/FormInputList/FormInputList.tsx]: Done

File: src/dex-ui-components/base/Icons/DefaultLogoIcon.tsx (hunk 3 above)
Comment (roshan-bisht-lab49) [reply src/dex-ui-components/base/Icons/DefaultLogoIcon.tsx]: Done, also same I updated for tooltipIcon

File: src/dex-ui/pages/dao/CreateADAO/forms/DAODetailsForm.tsx (hunk 4 above)
Comment (roshan-bisht-lab49) [reply src/dex-ui/pages/dao/CreateADAO/forms/DAODetailsForm.tsx]: My bad, its done.

File: src/dex-ui/pages/dao/CreateADAO/forms/MultiSigDAOGovernanceForm.tsx (hunk 5 above)
Comment (roshan-bisht-lab49) [reply src/dex-ui/pages/dao/CreateADAO/forms/MultiSigDAOGovernanceForm.tsx]: Done

File: src/dex-ui/utils/utils.ts (hunk 6 above)
Comment (roshan-bisht-lab49) [reply src/dex-ui/utils/utils.ts]: Done

**Review by damiano-melcarne-lab49 — APPROVED** [review]

**Pull request merged**